modules/internal/DefaultRectangular.chpl

```diff
@@ -1939,23 +1939,32 @@ module DefaultRectangular {
 
     const Aidx = A.getDataIndex(Alo);
     const Adata = _ddata_shift(A.eltType, A.theData, Aidx);
+    const Alocid = Adata.locale.id;
```

Comment:
Why cache this? I don't _think_ it would induce comm if it wasn't cached.

Reply:
(I'm ok with doing this, just curious about the motivation)

Reply:
Yeah I don't think it would cause comm otherwise. It is just a (premature?) optimization to avoid getting this ID again and again in this function and in the helpers.